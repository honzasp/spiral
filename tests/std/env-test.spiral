(program
  (import std)
  (import std.env)
  (import std.test)

  (test "env-get-argv" (lambda (t)
    (assert-equal t (env-get-argv)
      (cons "./std/env-test.exec"
      (cons "alan turing"
      (cons "alonzo church" null))))))

  (test "env-get-var" (lambda (t)
    (assert-eqv t (env-get-var "SPIRAL_TEST_ENV") "haskell curry")
    (assert-eqv t (env-get-var "MISSING_ENV_VAR") false))))
