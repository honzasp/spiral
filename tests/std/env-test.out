test env-get-argv ok
test env-get-var ok
